### ISSUE TITLE
Cannot use inits for a model with constrained types

### ISSUE THREAD

**Issue by wds15** [stan-dev/cmdstan#850]
#### Summary:
I am not able to start programs with constrained types when these go into a corner case. See below.

#### Description:

Take this dummy model:

```stan
data {
  int<lower=1> num;
}
parameters {
  real mu;
  cholesky_factor_corr[num] L_corr;
}
model {
  mu ~ std_normal();
  L_corr ~ lkj_corr_cholesky(1.0);
}
```

whenever now `num= 1` it is not possible to start this program with an init given to cmdstan. With this input to the init argument I am getting

```R
mu <- 0
L_corr <-
structure(integer(0),
.Dim = c(0, 0))
```

```
Unrecoverable error evaluating the log probability at the initial value.
Exception: matrix[uni,uni] indexing, row: accessing element out of range. index 1 out of range; container is empty and cannot be indexed (in '../OncoBayes2/mass-matrix-reuse/constrain_bug.stan', line 6, column 2 to column 35)
Exception: matrix[uni,uni] indexing, row: accessing element out of range. index 1 out of range; container is empty and cannot be indexed (in '../OncoBayes2/mass-matrix-reuse/constrain_bug.stan', line 6, column 2 to column 35)
```

when using this init of

```R
mu <- 0
L_corr <-
structure(c(1),
.Dim = c(1, 1))
```

I am getting

```
Unrecoverable error evaluating the log probability at the initial value.
Exception: matrix[uni,uni] indexing, row: accessing element out of range. index 1 out of range; container is empty and cannot be indexed (in '../OncoBayes2/mass-matrix-reuse/constrain_bug.stan', line 6, column 2 to column 35)
Exception: matrix[uni,uni] indexing, row: accessing element out of range. index 1 out of range; container is empty and cannot be indexed (in '../OncoBayes2/mass-matrix-reuse/constrain_bug.stan', line 6, column 2 to column 35)
```

I am using R data files as input to CmdStan.

#### Reproducible Steps:

See above.

#### Current Output:
Things don't work

#### Current Version:
v2.22.1

**Comment by rok-cesnovar** [comment]
See https://github.com/stan-dev/stanc3/issues/485

**Comment by wds15** [comment]
Thanks! Looks like this is a duplicate. This should really be fixed before the 2.23 goes out.

**Comment by rok-cesnovar** [comment]
> This should really be fixed before the 2.23 goes out.

Agreed. 

Hopefully someone is able to find a fix for this. I am going to close this here in order to keep the conversation in one place.
